### ISSUE TITLE
GoToDefinition jumps to unexpected place

### ISSUE THREAD

**Issue by nils-a** [OmniSharp/omnisharp-roslyn#2211]
My minimal example is 

```
public class Test
{
    public T MyFirst<T>(T[] t)
    {
        return Enumerable.First<T>(t);
    }
}
```

The point being here, that I call an extension method explicitly and not as an extension.

Now, positioning the cursor in `First` and calling `GoToDefinition` in VSCode results in the following request:

```
[dbug]: OmniSharp.Stdio.Host
        ************ Request ************
{
  "Type": "request",
  "Seq": 2671,
  "Command": "/v2/gotodefinition",
  "Arguments": {
    "FileName": "d:\\_dev\\test-2\\build\\Program.cs",
    "Line": 39,
    "Column": 31,
    "WantMetadata": true
  }
}
```

and in the response
```
[dbug]: OmniSharp.Stdio.Host
        ************  Response (3.1566ms) ************ 
{
  "Request_seq": 2671,
  "Command": "/v2/gotodefinition",
  "Running": true,
  "Success": true,
  "Message": null,
  "Body": {
    "Definitions": [
      {
        "Location": {
          "FileName": "$metadata$\\Project\\Build\\Assembly\\System\\Linq\\Symbol\\System\\Linq\\Enumerable.cs",
          "Range": {
            "Start": {
              "Line": 0,
              "Column": 0
            },
            "End": {
              "Line": 0,
              "Column": 0
            }
          }
        },
        "MetadataSource": {
          "AssemblyName": "System.Linq",
          "TypeName": "System.Linq.Enumerable",
          "ProjectName": "Build",
          "VersionNumber": null,
          "Language": null
        },
        "SourceGeneratedFileInfo": null
      }
    ]
  },
  "Seq": 66145,
  "Type": "response"
}
```

Notice how `Line` and `Column` are both `0`

However, I would have expected the result to be the same as if I were calling `GoToDefinition` on `(new string[0]).First()`.
I.e. I expect the result to be 

```
[dbug]: OmniSharp.Stdio.Host
        ************  Response (3.5806ms) ************ 
{
  "Request_seq": 2766,
  "Command": "/v2/gotodefinition",
  "Running": true,
  "Success": true,
  "Message": null,
  "Body": {
    "Definitions": [
      {
        "Location": {
          "FileName": "$metadata$\\Project\\Build\\Assembly\\System\\Linq\\Symbol\\System\\Linq\\Enumerable.cs",
          "Range": {
            "Start": {
              "Line": 1020,
              "Column": 30
            },
            "End": {
              "Line": 1020,
              "Column": 35
            }
          }
        },
        "MetadataSource": {
          "AssemblyName": "System.Linq",
          "TypeName": "System.Linq.Enumerable",
          "ProjectName": "Build",
          "VersionNumber": null,
          "Language": null
        },
        "SourceGeneratedFileInfo": null
      }
    ]
  },
  "Seq": 72789,
  "Type": "response"
}
```

**Comment by filipw** [comment]
are you able to share the repro project?

When I try to follow your steps, it works correctly, namely I get

```
{
  "Request_seq": 278,
  "Command": "/v2/gotodefinition",
  "Running": true,
  "Success": true,
  "Message": null,
  "Body": {
    "Definitions": [
      {
        "Location": {
          "FileName": "$metadata$/Project/net50-webapi/Assembly/System/Linq/Symbol/System/Linq/Enumerable.cs",
          "Range": {
            "Start": {
              "Line": 1245,
              "Column": 30
            },
            "End": {
              "Line": 1245,
              "Column": 35
            }
          }
        },
        "MetadataSource": {
          "AssemblyName": "System.Linq",
          "TypeName": "System.Linq.Enumerable",
          "ProjectName": "net50-webapi",
          "VersionNumber": null,
          "Language": null
        },
        "SourceGeneratedFileInfo": null
      }
    ]
  },
  "Seq": 709,
  "Type": "response"
}
```
![Bildschirmaufnahme 2021-08-22 um 21 34 31](https://user-images.githubusercontent.com/1710369/130367833-020228ab-e1f0-48d6-bdf2-f606e2dc4138.gif)

**Comment by nils-a** [comment]
@filipw interesting, I can no longer reproduce that, either.
Does it help if I promise, that I did saw this? 😬 

I started analyzing this, coming from https://github.com/OmniSharp/omnisharp-roslyn/pull/2212/files#diff-2f7ef0430bd563e1e334c985010d6715cd3f6a9ce6bf77839781eee8d22cbe33R118 and thought I'd found why the `GoToDefinition` results in position `0,0` when it did this in c# projects, too. 
But since whatever was the problem in c# then is gone now, I'll probably keep working on https://github.com/OmniSharp/omnisharp-roslyn/pull/2212/ for a bit.
